### PR TITLE
Cypress test for contact form in com_contact.

### PR DESCRIPTION
**Pull request by rajputanuj31**
Add a test for contact form in com_contact.

### CODE REVIEW

**Comment by laoneo** [comment]
Can you directly open the contact details page instead of going through the list?

File: tests/System/integration/site/components/com_contact/Form.cy.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+describe('Test that the Contact Form', () => {
+  it('can display a Contact Form', () => {
+    cy.task('queryDB', `SELECT id FROM #__users WHERE username = '${Cypress.env('username')}'`)
```
Comment (laoneo) [inline tests/System/integration/site/components/com_contact/Form.cy.js]: Can your create for this query a command as we are going to use functionality in the future on multiple places.

Comment (rajputanuj31) [reply tests/System/integration/site/components/com_contact/Form.cy.js]: Please Check it. Is it fine??

File: tests/System/integration/site/components/com_contact/Form.cy.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  it('can display a Contact Form', () => {
+    cy.db_getUserId().then((id) => {
+      cy.db_createContact({ name: 'contact 1', user_id: id, featured: 1 })
+        .then(() => {
+          cy.task('queryDB', 'SELECT id FROM #__contact_details WHERE name = \'contact 1\'')
+            .then(async (contactId) => {
+              cy.visit(`index.php?option=com_contact&view=contact&id='${contactId[0].id}'`);
+            });
+
+          cy.contains('Contact Form');
```
Comment (laoneo) [inline tests/System/integration/site/components/com_contact/Form.cy.js]: This two lines need to go after line 8 and then you can remove the async keyword.

File: tests/System/integration/site/components/com_contact/Form.cy.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+describe('Test that the Contact Form', () => {
+  it('can display a Contact Form', () => {
+    cy.db_getUserId().then((id) => {
+      cy.db_createContact({ name: 'contact 1', user_id: id, featured: 1 })
+        .then(() => {
+          cy.task('queryDB', 'SELECT id FROM #__contact_details WHERE name = \'contact 1\'')
```
Comment (laoneo) [inline tests/System/integration/site/components/com_contact/Form.cy.js]: This here is not needed as the contact id is passed to the then function which you made here empty, look here https://github.com/joomla/joomla-cms/blob/4.4-dev/tests/System/integration/api/com_contact/Contacts.cy.js#L28.

File: tests/System/support/commands/db.js [hunk 4]
```diff
@@ -142,3 +142,11 @@ Cypress.Commands.add('db_createUser', (userData) => {
     return info.insertId;
   });
 });
+
+Cypress.Commands.add('db_getUserId', () => {
+  cy.task('queryDB', `SELECT id FROM #__users WHERE username = '${Cypress.env('username')}'`)
+    .then(async (id) => {
+      
+      return cy.wrap(id[0].id);
```
Comment (laoneo) [inline tests/System/support/commands/db.js]: Do you need here the wrap?

File: tests/System/support/commands/db.js [hunk 5]
```diff
@@ -142,3 +142,11 @@ Cypress.Commands.add('db_createUser', (userData) => {
     return info.insertId;
   });
 });
+
+Cypress.Commands.add('db_getUserId', () => {
+  cy.task('queryDB', `SELECT id FROM #__users WHERE username = '${Cypress.env('username')}'`)
+    .then(async (id) => {
+
```
Comment (laoneo) [inline tests/System/support/commands/db.js]: I would add here a check if the id.length === 0, then return 0.

File: tests/System/integration/site/components/com_contact/Form.cy.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+describe('Test that the Contact Form', () => {
+  it('can display a Contact Form', () => {
+    cy.db_getUserId().then((id) => {
+      cy.db_createContact({ name: 'contact 1', user_id: id, featured: 1 })
+        .then((ContactId) => {
```
Comment (laoneo) [inline tests/System/integration/site/components/com_contact/Form.cy.js]: Can you put the variable into lower case, something like contactId?

**Comment by laoneo** [comment]
Can you also rename the file to Contact.cy.js as it is the contact details page, just showing the form.

File: tests/System/integration/site/components/com_contact/Form.cy.js (hunk 6 above)
Comment (rajputanuj31) [reply tests/System/integration/site/components/com_contact/Form.cy.js]: Done.

File: tests/System/integration/site/components/com_contact/Contact.cy.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+describe('Test that the Contact Form', () => {
+  it('can display a Contact Form', () => {
+    cy.db_getUserId().then((id) => {
```
Comment (laoneo) [inline tests/System/integration/site/components/com_contact/Contact.cy.js]: Remove the { and at the end the } then the return value of the expression will be returned, this is a shorthand of the arrow function.

Comment (rajputanuj31) [reply tests/System/integration/site/components/com_contact/Contact.cy.js]: like this ??
it('can display a Contact Form', () => {
`cy.db_getUserId().then((id) => cy.db_createContact({ name: 'contact 1', user_id: id, featured: 1 }))`
    .then((ContactId) => {

Comment (laoneo) [reply tests/System/integration/site/components/com_contact/Contact.cy.js]: Yes

File: tests/System/integration/site/components/com_contact/Contact.cy.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+describe('Test that the Contact Form', () => {
+  it('can display a Contact Form', () => {
+    cy.db_getUserId().then((id) => cy.db_createContact({ name: 'contact 1', user_id: id, featured: 1 }))
```
Comment (laoneo) [inline tests/System/integration/site/components/com_contact/Contact.cy.js]: is featured needed?

**Comment by laoneo** [comment]
Thanks!

**Comment by rajputanuj31** [comment]
Thanks a lot for merging it.

**Pull request merged**